modules/components/src/AdvancedFacetView/SearchBox.js

```diff
@@ -50,20 +146,44 @@ export default class extends React.Component {
             this.setState({ isDropdownShown: true });
             e.stopPropagation();
           }}
-          onChange={e => this.setState({ currentValue: e.target.value })}
+          onChange={e =>
+            this.setState(
+              {
+                isDropdownShown: true,
+                currentValue: e.target.value,
+              },
+              () => {
+                const newFilteredList = this.getFilteredList();
+                this.setState({
+                  highlightedField: newFilteredList?.[0]?.field,
+                });
+              },
+            )
+          }
+          onKeyDown={e => this.handleKeyPress(e)}
         />
         {filteredList?.length && isDropdownShown ? (
           <div className={`resultList shown`}>
             {filteredList?.map(({ displayName, field, ...rest }) => (
               <div
                 key={field}
-                className="resultItem"
+                ref={el => (this.dropdownRefs[field] = el)}
+                onMouseEnter={e => this.setState({ highlightedField: field })}
+                className={`resultItem ${
+                  highlightedField === field ? 'highlighted' : ''
+                }`}
                 onClick={() => {
+                  this.setState({ currentValue: displayName });
                   onFieldSelect(field);
                 }}
               >
-                <span className="title">{displayName}</span>
-                <span className="field">{field}</span>
+                <span
+                  className="title"
+                  dangerouslySetInnerHTML={{
```

Comment:
I'm 99% sure we shouldn't be doing this. Why not just pass a regular react component?